common/src/main/java/com/microsoft/identity/common/internal/cache/SchemaUtil.java

```diff
@@ -137,6 +138,38 @@ public static String getAlternativeAccountId(final IDToken idToken) {
         return alternativeAccountId;
     }
 
+    public static String getIdentityProvider(final String idTokenString) {
+        final String methodName = "getIdentityProvider";
+
+        String idp = null;
+
+        if (null != idTokenString) {
+            IDToken idToken;
+            try {
+                idToken = new IDToken(idTokenString);
+                final Map<String, String> idTokenClaims = idToken.getTokenClaims();
+
+                if (null != idTokenClaims) {
+                    idp = idTokenClaims.get("idp");
```

Comment:
There should be constant somewhere for this, probably inside of `AuthenticationConstants` or one of the [many] `IdToken` classes